opencti-platform/opencti-graphql/src/utils/stix-filtering/stix-testers.ts

```diff
@@ -0,0 +1,378 @@
+    const extendedStixValues: string[] = [...toValidArray(stixValue), ...getParentTypes(stixValue)];
+    return testStringFilter(filter, extendedStixValues);
+  }
+  return false;
+};
+
+/**
+ * RELATION TO TYPES
+ * - depending on stix type (relation or sighting), we might search in target_type or where_sighted_types (in extension)
+ * - we must also search in parent types
+ */
+export const testRelationToTypes = (stix: any, filter: Filter) => {
+  if (stix.type === STIX_TYPE_RELATION) {
+    const stixValue: string = stix.extensions?.[STIX_EXT_OCTI].target_type;
+    const extendedStixValues: string[] = [...toValidArray(stixValue), ...getParentTypes(stixValue)];
+    return testStringFilter(filter, extendedStixValues);
+  }
+  if (stix.type === STIX_TYPE_SIGHTING) {
+    const stixValues: string[] = stix.extensions?.[STIX_EXT_OCTI].where_sighted_types ?? [];
+    const extendedStixValues = [...stixValues, ...stixValues.map((t) => getParentTypes(t)).flat()];
+    return testStringFilter(filter, extendedStixValues);
+  }
+  return false;
+};
+
+/**
+ * REFS
+ * - uses a refs extractor using the standardId generator
+ */
+export const testRefs = (stix: any, filter: Filter) => {
+  const stixValues: string[] = stixRefsExtractor(stix, generateStandardId);
+  return testStringFilter(filter, stixValues);
+};
+
+/**
+ * INSTANCE TYPE (elementId)
+ * - elementId is id in stix extension
+ * - useSideEventMatching arg to optionally test against relations and refs of the object
+ */
+export const testInstanceType = (stix: any, filter: Filter, useSideEventMatching = false) => {
+  const stixValues: string[] = toValidArray(stix.extensions?.[STIX_EXT_OCTI]?.id);
+  if (!useSideEventMatching) {
+    // basic equality test between ids
+    return testStringFilter(filter, stixValues);
+  }
+
+  // useSideEventMatching is set ; only applies with "eq" operator
+  if (filter.operator !== 'eq') {
+    return false;
+  }
+
+  // advanced test between filter ids and the entity relations and refs
+  // we shall aggregate all candidate fields and match the filter
+  const aggregatedStixValues = [];
+  if (stix.type === STIX_TYPE_RELATION) {
+    aggregatedStixValues.push(...toValidArray(stix.target_ref)); // to
+    aggregatedStixValues.push(...toValidArray(stix.source_ref)); // from
+  }
+  if (stix.type === STIX_TYPE_SIGHTING) {
+    aggregatedStixValues.push(...(stix.where_sighted_refs ?? [])); // to
+    aggregatedStixValues.push(...toValidArray(stix.sighting_of_ref)); // from
+  }
+  // refs
+  aggregatedStixValues.push(...stixRefsExtractor(stix, generateStandardId));
+
+  return testStringFilter(filter, aggregatedStixValues);
+};
+
+/**
+ * Gives the right tester function according to the filter key.
+ * If the key is not handled, returns a function that always return false.
+ * TODO: make it dependent on the schema.
+ * @param key
+ */
+export const getStixTesterFromFilterKey = (key: string): TesterFunction => {
+  switch (key) {
+    case MARKING_FILTER:
```

Comment:
What about a mapping instead of a switch case ?
Like : 
```
{
  MARKING_FILTER: testMarkingFilter,
  TYPE_FILTER: testEntityType
 [...]
}

```

Reply:
yes, probably nicer. I'll do that !